packages/components/src/font-size-picker/index.tsx

```diff
@@ -46,61 +99,61 @@ const UnforwardedFontSizePicker = (
 		__next40pxDefaultSize = false,
 		fallbackFontSize,
 		fontSizes = [],
-		disableCustomFontSizes = false,
 		onChange,
+		pickerMode = 'both',
 		size = 'default',
 		units: unitsProp = DEFAULT_UNITS,
 		value,
 		withSlider = false,
 		withReset = true,
+
+		// deprecated
+		disableCustomFontSizes,
 	} = props;
 
-	const units = useCustomUnits( {
-		availableUnits: unitsProp,
-	} );
+	let computedPickerMode = pickerMode;
+	if ( disableCustomFontSizes !== undefined ) {
+		deprecated(
+			'`disableCustomFontSizes` prop in wp.components.FontSizePicker',
+			{
+				since: '6.7',
+				version: '6.9',
+				alternative: '`pickerMode` prop',
+			}
+		);
+
+		computedPickerMode = disableCustomFontSizes ? 'predefined' : 'both';
+	}
 
-	const shouldUseSelectControl = fontSizes.length > 5;
 	const selectedFontSize = fontSizes.find(
 		( fontSize ) => fontSize.size === value
 	);
 	const isCustomValue = !! value && ! selectedFontSize;
 
-	const [ showCustomValueControl, setShowCustomValueControl ] = useState(
-		! disableCustomFontSizes && isCustomValue
+	const [ currentPickerType, setCurrentPickerType ] = useState(
+		getPickerType( computedPickerMode, isCustomValue, fontSizes )
 	);
 
-	const headerHint = useMemo( () => {
```

Comment:
Moved `headerHint` computation to an external util function to make the render function easier to read